File: src/main/kotlin/customfit/ai/kotlinclient/client/FeatureFlagChangeListener.kt
package customfit.ai.kotlinclient.client

/**
 * An interface for objects that will be notified when a feature flag's value changes.
 * Register instances of this with [CFClient.registerFeatureFlagListener].
 */
interface FeatureFlagChangeListener {
    /**
     * Called when the value of a feature flag changes.
     *
     * @param flagKey the flag that changed
     * @param newValue the new value of the flag
     */
    fun onFeatureFlagChange(flagKey: String, newValue: Any)
} 

File: customfit-flutter-client-sdk/lib/src/client/listener/feature_flag_change_listener.dart
/// Listener for feature flag changes.
///
/// This interface is used to listen for changes to feature flags.
abstract class FeatureFlagChangeListener {
  /// Called when a feature flag value changes.
  ///
  /// [flagKey] is the key of the flag that changed.
  /// [oldValue] is the previous value of the flag.
  /// [newValue] is the new value of the flag.
  void onFeatureFlagChanged(String flagKey, dynamic oldValue, dynamic newValue);
}
